packages/headless/src/features/attached-results/attached-results-slice.test.ts

```diff
@@ -0,0 +1,160 @@
+import {createMockAttachedResult} from '../../test/mock-attached-results';
+import {
+  attachResult,
+  detachResult,
+  setAttachedResults,
+} from './attached-results-actions';
+import {attachedResultsReducer} from './attached-results-slice';
+import {
+  AttachedResultsState,
+  getAttachedResultsInitialState,
+} from './attached-results-state';
+
+describe('attached results slice', () => {
+  let state: AttachedResultsState;
+
+  beforeEach(() => {
+    state = getAttachedResultsInitialState();
+  });
+
+  it('initializes the state correctly', () => {
+    const finalState = attachedResultsReducer(undefined, {type: ''});
+    expect(finalState).toEqual({
+      results: [],
+    });
+  });
+
+  it('#setAttachedResults sets the state #results correctly', () => {
+    const testAttachedResult = createMockAttachedResult();
+    const action = setAttachedResults({
+      results: [testAttachedResult],
+    });
+    const finalState = attachedResultsReducer(state, action);
+    expect(finalState.results).toBe(action.payload.results);
+    expect(finalState.results.length).toEqual(1);
+  });
+
+  it('#setAttachedResults accepts an empty array of results', () => {
+    const action = setAttachedResults({
+      results: [],
+      message: 'foo',
+    });
+    const finalState = attachedResultsReducer(state, action);
+    expect(finalState).toStrictEqual(action.payload);
+    expect(finalState.results.length).toEqual(0);
+  });
+
+  it('#setAttachedResults correctly adds the result as attach so we can find it later', () => {
```

Comment:
I can't see the difference between this test and the `#setAttachedResults sets the state #results correctly` test, can you please elaborate?

Reply:
In the second test I'm testing that after setting the results, we can actually find the result we've just set.
In the first test, I test the action payload and the number of results that were set.

I guess they are very similar but slightly different. 🤷

Reply:
I merged both tests